File: ssort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h> 

#define RANGE_MAX 1000

void print_list(int* array, int n)
{
   for (int i = 0; i < n; i++)
      printf("%d\t", array[i]);
   printf("\n");
}

void random_elements(int* array, int n) 
{
   /* Intializes random number generator */
   srand(time(0));

   for (int i = 0; i < n; i++)
      array[i] = rand() % RANGE_MAX;
}

void selection_sort(int* array, int n)
{
   int i, j, min_pos, tmp;

   // finding minimum element (n-1) times
   for (i = 0; i < (n - 1); i++) {
      min_pos = i;
      for (j = i + 1; j < n; j++) {
         if (array[min_pos] > array[j])
            min_pos = j;
      }
      if (min_pos != i) {
         tmp = array[i];
         array[i] = array[min_pos];
         array[min_pos] = tmp;
      }
   }
}

int main(int argc, char *argv[])
{
   int n = 10;
   if (argc == 2)
      n = atoi(argv[1]);

   //printf("Enter number of elements: ");
   //scanf("%d", &n);
   
   int* array = (int *) malloc(n * sizeof(int));
   
   random_elements(array, n);

   printf("List before sorting:\n");
   print_list(array, n);

   selection_sort(array, n);

   printf("Sorted list in ascending order:\n");
   print_list(array, n);

   return 0;
}

File: ssort.py
import random
import sys

RANGE_MAX = 1000

def print_list(array, n):
   for i in range(n):
      print(str(array[i]) + '\t', end='')
   print()


def random_elements(array, n):
   for i in range(n):
      array[i] = random.randrange(RANGE_MAX)


def selection_sort(array, n):
   # finding minimum element (n-1) times
   for i in range(n - 1):
      min_pos = i
      for j in range(i + 1, n):
         if array[min_pos] > array[j]:
            min_pos = j
      if min_pos != i:
         tmp = array[i]
         array[i] = array[min_pos]
         array[min_pos] = tmp


if __name__ == "__main__":
   n = 10
   if len(sys.argv) == 2:
      n = int(sys.argv[1])
   #print("n = " + str(n))
   array = [0] * n
   
   random_elements(array, n)

   print("List before sorting:")
   print_list(array, n)

   selection_sort(array, n)

   print("Sorted list in ascending order:")
   print_list(array, n)

